repositories/convoy_repository.go

```diff
@@ -28,16 +29,26 @@ func (noOpConvoyClientProvider) GetProjectID() string {
 	return ""
 }
 
-func NewConvoyRepository(convoyClientProvider ConvoyClientProvider) ConvoyRepository {
+type ConvoyRepository struct {
+	convoyClientProvider ConvoyClientProvider
+	limiter              *rate.Limiter
+}
+
+func NewConvoyRepository(convoyClientProvider ConvoyClientProvider, limit int) ConvoyRepository {
 	if convoyClientProvider == nil {
 		convoyClientProvider = noOpConvoyClientProvider{}
 	}
 
-	return ConvoyRepository{convoyClientProvider: convoyClientProvider}
-}
-
-type ConvoyRepository struct {
-	convoyClientProvider ConvoyClientProvider
+	// other instances may be running in parallel. If we want to do it more precisely, we'll need a distributed limiter
+	instanceLimit := rate.Limit(limit) / 2
+	burst := limit / 3
+	if burst < 1 {
```

Comment:
burst = 0 means no events sent (unless limit = infinity by convention)

Reply:
Can you change for explicit equal 0 condition instead of inequality ?